ciao-launcher/main.go

```diff
@@ -356,6 +353,28 @@ DONE:
 			if err != nil {
 				break DONE
 			}
+			clusterConfig,err := client.conn.ClusterConfiguration()
+			if err != nil {
+				glog.Errorf("Unable to get Cluster Configuration %v", err)
+				dialCh <- err
```

Comment:
Is this a copy and paste error?  Maybe it was in the code I sent you.  I think this will deadlock.  There's no one to read from dialCh.  We can just remove this line.